arch/sim/src/sim/up_idle.c

```diff
@@ -116,4 +116,8 @@ void up_idle(void)
 
   up_timer_update();
 #endif
+
+#ifdef CONFIG_SIM_HCISOCKET
+  bthcisock_read();
```

Comment:
change to bthcisock_loop to align with other similiar function better(.e.g. netdriver_loop)?

Reply:
I changed it to read from loop because it matches more of what it is and it is also called internally where loop would be confusing.

Reply:
But if you look at the prototype:
```
int bthcisock_read(void);
```
it doesn't like the normal read function which accept the buffer and length argument:
```
int bthcisock_host_read(int fd, uint8_t *type, void *buf, size_t len);
```
And in the implementation, it call bthcisock_host_read and bt_hci_receive which convert the pull model(read) to the push model(receive).
So from both the prototype and implementation, read isn't a suitable name.

Reply:
What about `bthcisock_receive` since at the end it's calling receive in the bt stack?

Reply:
The key point is that the function called by idle loop can't accept any argument, and then is very strange to name it to read or receive because these functions normally accept a buffer argument to indicate where to read/receive the data.

Reply:
Made it loop, I don't like the name but that can be addressed when we eventually can have worker threads instead of the idle loop.